### PR TITLE
`OrtResult` function renamings

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Rename several `collect...` functions to `get...` and do some minor refactorings. See the commit messages for details.

**Breaking API change**:
Several `OrtResult` functions have changed their name.

### CODE REVIEW

File: model/src/main/kotlin/OrtResult.kt [hunk 1]
```diff
@@ -96,35 +96,22 @@ data class OrtResult(
         )
     }
 
-    /** An object that can be used to navigate the dependency information contained in this result. */
```
Comment (sschuberth) [inline model/src/main/kotlin/OrtResult.kt:99]: Commit message nit: Is sorting really refactoring? Maybe rather "chore" / "style"?

File: model/src/main/kotlin/OrtResult.kt [hunk 2]
```diff
@@ -163,6 +163,30 @@ data class OrtResult(
 
     private val scanResultsById: Map<Identifier, List<ScanResult>> by lazy { scanner?.scanResults.orEmpty() }
 
+    /**
```
Comment (sschuberth) [inline model/src/main/kotlin/OrtResult.kt:166]: Same commit message nit here about the "refactor" type.

**Review by sschuberth — COMMENTED** [review]
Do you also want to address `collectedIssues` in `DependencyGraph` / `DependencyNavigator` for consistency?

**Comment by mnonnenmacher** [comment]
> Do you also want to address `collectedIssues` in `DependencyGraph` / `DependencyNavigator` for consistency?

Yes, but not in this PR, I wanted to focus on `OrtResult` first.

File: model/src/main/kotlin/OrtResult.kt (hunk 1 above)
Comment (mnonnenmacher) [reply model/src/main/kotlin/OrtResult.kt:99]: I think "chore" fits best.

**Review by sschuberth — APPROVED** [review]
I'm fine with this, the `collect*` function names have bothered me before, too. I'm just a bit concerned about how many rule scripts this might break.

**Comment by mnonnenmacher** [comment]
> I'm fine with this, the `collect*` function names have bothered me before, too. I'm just a bit concerned about how many rule scripts this might break.

Rule scripts should not be affected as long as the DSL is used, it's more likely notifier scripts like the example script, but I think the notifier is not being used that much.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**